### PR TITLE
Remove es5 cc class

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: cocos-creator/3d-tasks#

Changelog:
 * Remove `CCClass()` style cc-class declaration.

@jareguo  Please review this.

In detail, I removed:

- `CCClass(options)`'s `options.__es6__`, `options.__ctor__`, `options.statics`.

### CODE REVIEW

File: cocos/core/data/class.ts [hunk 1]
```diff
@@ -257,84 +218,30 @@ function mixinWithInherited (dest, src, filter?) {
...
-                        shouldAddProtoCtor = true;
-                    }
-                }
-            }
-            if (ctor) {
-                if (__ctor__) {
-                    errorID(3649, className);
-                }
-                else {
-                    ctor = options.ctor = _validateCtor_DEV(ctor, baseClass, className, options);
```
Comment (shrinktofit) [inline cocos/core/data/class.ts:291]: super call is checked by ES6 class.

File: cocos/core/data/class.ts [hunk 2]
```diff
@@ -919,51 +434,21 @@ export function CCClass (options) {
...
             deferredInitializer.push({ cls, props: properties, mixins });
             cls.__props__ = cls.__values__ = null;
         }
     }
     else {
-        declareProperties(cls, name, properties, base, options.mixins, options.__ES6__);
-    }
-
-    // define statics
-    const statics = options.statics;
```
Comment (shrinktofit) [inline cocos/core/data/class.ts:935]: Statics can be expressin in ES6 class syntax

File: cocos/core/data/class.ts [hunk 3]
```diff
@@ -121,30 +118,10 @@ function appendProp (cls, name) {
...
     const defaultValue = val.default;
 
     if (DEV) {
-        if (!es6) {
-            // check default object value
-            if (typeof defaultValue === 'object' && defaultValue) {
-                if (Array.isArray(defaultValue)) {
-                    // check array empty
-                    if (defaultValue.length > 0) {
-                        errorID(3635, className, propName, propName);
```
Comment (jareguo) [inline cocos/core/data/class.ts:134]: 这些用不到的 message ID，需要在 EngineErrorMap.md 里对应的字段上加上 `<!-- DEPRECATED -->`，这样引擎在 dev 模式下就不会再读取到这些废弃的字符串，节省内存。例如：
image

File: cocos/core/data/class.ts [hunk 4]
```diff
@@ -919,51 +434,21 @@ export function CCClass (options) {
...
-        }
-        // use value to redefine some super method defined as getter
-        js.value(cls.prototype, funcName, func, true, true);
+        declareProperties(cls, name, properties, base, options.mixins);
     }
 
     const editor = options.editor;
     if (editor) {
-        if (js.isChildClassOf(base, legacyCC.Component)) {
+        if (typeof base === 'function' && js.isChildClassOf(base, legacyCC.Component)) {
```
Comment (jareguo) [inline cocos/core/data/class.ts]: 感觉这个 typeof 不太需要？isChildClassOf 里面已经有判断了

File: cocos/core/data/class.ts (hunk 3 above)
Comment (shrinktofit) [reply cocos/core/data/class.ts:134]: https://github.com/cocos-creator/engine/pull/7504/commits/06fb69456dcef404001a1820d7b9bc70e79f189e

File: cocos/core/data/class.ts (hunk 4 above)
Comment (shrinktofit) [reply cocos/core/data/class.ts]: https://github.com/cocos-creator/engine/pull/7504/commits/06fb69456dcef404001a1820d7b9bc70e79f189e

**Review by jareguo — APPROVED** [review]

**Pull request merged**